### DISCUSSION TITLE
Our Service Education- Level 1

### DISCUSSION THREAD

**Discussion by zsamji** [bcgov/sso-keycloak#136] [Ideas]
Welcome to our first installment of learning all things Pathfinder SSO. 

This section is lovingly called Level 1 and acts as an introduction to concepts and terms as you make use of our Self Service application to integrate your digital application with a with BC government approved login option. 

1.  To view the Identity providers, aka login options, that we provide, visit the section on ["Which Identity Provider Should I use" ](https://bcgov.github.io/sso-requests/faq#q-which-identity-provider-should-i-use )

2.   To understand which technical frameworks that you are currently using in might be using for your application, whether it is a single page apps, multipage apps, or servers side, review this [link]( https://www.mindinventory.com/blog/best-backend-frameworks/ ): and [this one](https://www.monocubed.com/blog/top-single-page-application-frameworks/)
3.   Learn more about [our self-service and its technology stack]( https://github.com/bcgov/sso-keycloak/wiki/What-is-Keycloak-@-BC-Government%3F) here:
   - Understand why we are promoting [Standard Realms:](https://github.com/bcgov/sso-keycloak/wiki/What-is-Keycloak-@-BC-Government%3F#why-a-standard-realm)
4.   To learn more about Protocols SAML,  and OIDC and OAuth Types of Authentication & Authorization protocols review the links below:
   -  [User Authentication  ](https://www.parallels.com/blogs/ras/oauth-vs-saml-vs-openid/#:~:text=What%20Are%20the%20Differences%20between,are%20used%20in%20user%20authentication.)
   -  [openid, oauth and saml](https://www.okta.com/identity-101/whats-the-difference-between-oauth-openid-connect-and-saml/)
   -  [Okta](https://developer.okta.com/blog/2017/07/25/oidc-primer-part-1)
- To understand endpoints, please review the bullet above on protocols, as well [as this link]( https://github.com/bcgov/sso-keycloak/wiki/test-endpoints):

Explore more in our [Wiki ](https://github.com/bcgov/sso-keycloak/wiki)

Stay tuned as we create a [new discussion thread on Level 2](https://github.com/bcgov/sso-keycloak/discussions/137) that dives deeper into tokens and example applications.

**Comment by zsamji** [comment]
feedback....
Part 1 okta link i find the hardest to read as theres a lot of technical jargon (i think this might be suited for level 2); okta 101 was also a bit tricky but the examples at the end were really helpful - so maybe we can link directly to those?; parallels link was easiest one for me to read and i think is good for a level 1 :)

so maybe we do an “overview” of saml and oidc in level 1 with the parallels link, and then add the 2 okta links to level 2?

also can we make the links open in a new page.

**Reply by kurokn** [reply]
answer to the last question: looks like we are not able to do that
refer: https://github.com/mojombo/github-flavored-markdown/issues/28
issue which started in 2012 and still keep on...

**Comment by thegentlemanphysicist** [comment]
Jon's notes:
 - in bullet point 1 the link mentions needing a redirect URI in the question Which info do I need to get started, that's contradicted later on the page when it says that you can use local host. I'd put an `(optional but recomended)` next to that entry
 - in bullet point 2 'tecnical' is repeated, also link only discusses server side rendered apps 
 - bullet point 3 has  deadlink in the first paragraph.  I would probably put this entry first in the list of resources to read. I believe adding a link to explain what OIDC is the first time it's mentioned in this page would be useful
 - bullet point 4 seems redundant with 'what is keycloak' bullet point.  The diagram is in both links

**Reply by zsamji** [reply]
Thx


Re bullet 1:  I'm not following the recommendation on optional but recommended ). the intent is for the audience to read the identity provider section only

![image](https://user-images.githubusercontent.com/56739669/177889165-727b9059-185a-434f-92e2-dba0741037d9.png)

**Reply by zsamji** [reply]
do you have a suggested article for non server side apps? re: in bullet point 2 'tecnical' is repeated, also link only discusses server side rendered apps

**Reply by zsamji** [reply]
@tzhang200 @junminahn re comment above  "link only discusses server side rendered apps"...any links for non server side....Note this tied to bullet 2 ie "To understand which Technical frameworks that you...". thx

**Comment by tzhang200** [comment]
@zsamji we have a SPA app in level 2. This might be a good overview on SPA and frontend framework though https://www.monocubed.com/blog/top-single-page-application-frameworks/

**Comment by kurokn** [comment]
- feeling that all the learning materials provides a wide scope in understanding the basic concepts as a LEVEL 1 learning;
- materials of definitions and differences on those 3 protocals is the most difficult part for this level, but can depend on readers choice on what to learn;

**Reply by kurokn** [reply]
also, typo on [word "availble" in wiki](https://github.com/bcgov/sso-keycloak/wiki/What-is-Keycloak-@-BC-Government%3F)
and some in this main reference comment, not sure if these matters

**Reply by zsamji** [reply]
sorry the tardiness but thanks for your feedback; Based on your feedback, I added a key term section and  we will need a way to keep the list of terms updated https://github.com/bcgov/sso-keycloak/wiki/Useful-References#intro-to-terms
